packages/vite/src/node/cleanString.ts

```diff
@@ -0,0 +1,31 @@
+// bank on the non-overlapping nature of regex matches and combine all filters into one giant regex
+// /`([^`\$\{\}]|\$\{(`|\g<1>)*\})*`/g can match nested string template
+// but js not support match expression(\g<0>). so clean string template(`...`) in other ways.
+const cleanerRE = /"[^"]*"|'[^']*'|\/\*(.|[\r\n])*?\*\/|\/\/.*/g
+
+const blankReplacer = (s: string) => ' '.repeat(s.length)
+const stringBlankReplacer = (s: string) =>
+  `${s[0]}${'\0'.repeat(s.length - 2)}${s[0]}`
+
+export function emptyCommentsString(raw: string): string {
+  return raw.replace(cleanerRE, (s: string) =>
+    s[0] === '/' ? blankReplacer(s) : s
+  )
+}
```

Comment:
I also noticed that this function isn't used. Is this in preparation for a future PR?

Reply:
I think yes, but I'm not sure it can use in https://github.com/vitejs/vite/issues/7549.